### ISSUE TITLE
Slides for Introduction

### ISSUE THREAD

**Issue by joshcho** [ekmett/lens#980]
This might not be the right place (feel free to close). But I am just learning about Lens, and the laws on the first slide of [http://comonad.com/haskell/Lenses-Folds-and-Traversals-NYC.pdf](introductory slides) were quite confusing (i.e. types don't match for the third law) . I ended up looking around and noticing [http://oleg.fi/gists/posts/2018-12-12-find-correct-laws.html](this blog) which corrects the third law.

If I am not mistaken, this could be a huge turn-off for beginners, especially if Lenses are known to be difficult.

**Comment by phadej** [comment]
the slides have

```
set :: Lens s a -> s -> a -> s
```

my post has both,btw.

```
set l v' (set l v s) ≡ set l v' s -- without type of set stated, but probably set :: Lens s a -> a -> s -> s
set (set s a) a' ≡ set s a' -- using set :: Lens s a -> s -> a -> s
```

**Comment by joshcho** [comment]
I guess I mean the first law here then.
```
set l (view l s) s = s
```
This doesn't quite make sense to me. Shouldn't it be:
```
set l s (view l s) = s
```
since `(view l s)` returns the element `a`.

**Comment by phadej** [comment]
@joshcho you are right, the `set` in first one has flipped arguments. I wonder if @ekmett still has editable slides somewhere.

**Comment by ekmett** [comment]
I'll hunt around for them.
